### PR TITLE
Feature/optimization et all in database

### LINKED ISSUE #87: Make it so that glitched and optimized no longer require a role, and are instead handled by the database variable.

### LINKED ISSUE #90: Make it so that a drone has to preface its messages with its ID, and that this is something that can be toggled on or off.
This value will then sit in the database. This will be toggled with the 'toggle_ID_prefacing_enforcement'.

### PR DESCRIPTION
**Pull request by HexCorpDrone**
#87 fix
#90 fix

Moves optimization, glitching, and the newly-added "id_prepending" all to the DB. Roles are still given and taken when toggling, but that's just so you can easily see which drones are set with what. Yay!

- New "is_optimized", "is_glitched", and "is_prepending_id" funcs to the drone DAO. Returns true or false depending on parameters in the DB.
- New on_message listener that checks if the author needs to prepend their message with their ID and the ::, and deletes the message if this check fails. 
- Speech optimization and glitching have been updated to read from the database rather than checking roles.

### CODE REVIEW

File: ai.py [hunk 1]
```diff
@@ -100,17 +103,27 @@ async def amplify(context, message: str, target_channel: discord.TextChannel, *d
                 await target_webhook.send(message, username=amp_profile["username"], avatar_url=amp_profile["avatar_url"])
 
 
+@bot.command(aliases=['tid'], brief="Hive Mxtress", usage=f'{bot.command_prefix}tip 5890 9813')
```
Comment (digitalAmy) [inline ai.py]: typo
```suggestion
@bot.command(aliases=['tid'], brief="Hive Mxtress", usage=f'{bot.command_prefix}tid 5890 9813')
```

File: ai.py [hunk 2]
```diff
@@ -100,17 +103,27 @@ async def amplify(context, message: str, target_channel: discord.TextChannel, *d
...
+async def toggle_id_prepending(context, *drones):
+
```
Comment (digitalAmy) [inline ai.py]: documentation for help texts missing

File: ai.py [hunk 3]
```diff
@@ -100,17 +103,27 @@ async def amplify(context, message: str, target_channel: discord.TextChannel, *d
                 await target_webhook.send(message, username=amp_profile["username"], avatar_url=amp_profile["avatar_url"])
```
Comment (digitalAmy) [inline ai.py]: Is this command guild_only or does it have any other restrictions?

**Review by digitalAmy — CHANGES_REQUESTED** [review]
A few smol issues with documentation should be addressed.

**Review by digitalAmy — CHANGES_REQUESTED** [review]
Still two open issues.

**Review by 3287 — CHANGES_REQUESTED** [review]
The very cute 9813 has already pointed out everything it's found issues with. Otherwise, wonderful job as always, 5890.

**Review by 3287 — APPROVED** [review]
Yay! Everything looks like it's been fixed. Good job, cutie drone.

**Review by digitalAmy — APPROVED** [review]
Very well done. One could say, perfectly done. Just like the drone that wrote this code is perfectly cute and good~

**Pull request merged**